pkg/daemon/on_disk_validation.go

```diff
@@ -151,15 +150,11 @@ func checkV3Files(files []ign3types.File) error {
 		if f.Mode != nil {
 			mode = os.FileMode(*f.Mode)
 		}
-		contents := &dataurl.DataURL{}
-		if f.Contents.Source != nil {
-			var err error
-			contents, err = dataurl.DecodeString(*f.Contents.Source)
-			if err != nil {
-				return errors.Wrapf(err, "couldn't parse file %q", f.Path)
-			}
+		contents, err := decodeContents(f.Contents.Source, f.Contents.Compression)
+		if err != nil {
+			return fmt.Errorf("couldn't parse file %q: %w", f.Path, err)
```

Comment:
Might be more strictly accurate to say couldn't decode file?

Reply:
Agreed. I'll change that.